src/common/recordbatch/src/recordbatch.rs

```diff
@@ -51,6 +55,24 @@ impl RecordBatch {
         })
     }
 
+    pub fn try_project(&self, indices: &[usize]) -> Result<Self> {
+        let schema = Arc::new(self.schema.try_project(indices).context(DataTypesSnafu)?);
```

Comment:
Maybe define a new error variant so that it can carry the table original schema and projection indices for troubleshooting.